core/service/manager.go

```diff
@@ -230,6 +233,35 @@ func (manager *Manager) List() map[ID]*Instance {
 	return manager.servicePool.List()
 }
 
+// ListAll returns array of all services
+func (manager *Manager) ListAll() []*Instance {
```

Comment:
There are many reasons to use non pointer slices. Less memory, fewer allocations needed and it's also faster. Pointer slices also serve little practical use, especially in this case here. It's almost always more trouble than its worth.

I see it's been done here like that before, am I just missing something?

Reply:
Actually yes, there's similar method called List, which also uses pointer instead of struct. So, I decided to go this way